### DISCUSSION TITLE
Success on TP-LINK Archer T9UH AC1900 V2 on Kali 6.8.11-1kali2 (amd64) (2024-05-30) Virtual Machine

### DISCUSSION THREAD

**Discussion by KyleDev008** [morrownr/8814au#144] [General]
Some context. I bought the TP-LINK Archer T9UH AC1900 V2 adapter because I won a CTF contest at work and got to choose a prize for it. After some reding online about which adapters support both 2.4 and 5 GHz and making sure that it had the correct chipset for monitor mode and packet injection, I settled on this one due to the support and the design that was more compact than having a long aerial. The purpose for it was to use it for learning about Info Sec with something I could potentially continue to use in the future on real life pen tests.

As soon as I received it yesterday, I tried using like I had used another cheaper adapter to start  practicing, and I immediately couldn't get a wifi connection and couldn't switch modes, because the adapter wasn't registering. I spent about 3 hours yesterday trying to go through the docs on the official site, some other repos, and all of them failed, and the one attempt actually bricked my 2nd kali distro by Zsecurity. So I'm retrying that one now, as I was following a course by them that requires that version of Kali (which by the way is visually nicer than the default kali for me), so I'll update here if it goes well.

But bottom line, thank you for this repo, I managed to get the driver installed and the adapter working and running on airodump-ng as a test and it worked!

**Comment by morrownr** [comment]
@KyleDev008 

A few words: I have a list of adapters recommended for use with Kali. You can go to the site Main Menu:

https://github.com/morrownr/USB-WiFi

Menu item 2 is the Plug and Play List and under it is the list for Kali.

Note that the list does not contain any adapters with Realtek chips. There is a reason for that. With this adapter based on the rtl8814au chip, you will get frustrated.

Let me know if you have any questions.

**Comment by KyleDev008** [comment]
Thanks for the feedback @morrownr . I'll definitely switch the adapter out for another non-realtek one down the line. For now this one should be good for learning the basics since I'm not planning on doing a kernel update for the next while on my kali machine and I can't send it back for a refund either. 

I actually went more for the design and the 4 antenna feature they advertise, and the fact that I found a couple sources online that said it does work for what I was looking for. But no one made it clear that the RealTek chips are an issue, which is what brought me to your repo in the first place. 

If I run into any issues and manage to solve them, I'll definitely post them here to help out those that also made the mistake of buying a RealTek adapter.

**Comment by morrownr** [comment]
> and the fact that I found a couple sources online that said it does work for what I was looking for.

Here is a link to a message that I posted in July to once again warn Kali users to be careful:

https://github.com/morrownr/USB-WiFi/issues/474

This driver is basically in maintenance mode which means I have no plans to do anything beyond accounting for basic API changes. Do not expect me to work bugs.

Regards
